opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -124,48 +125,137 @@ message MetricDescriptor {
+
+    // Temporality is the temporal quality values of a metric have.
+    Temporality temporality = 2;
+
+    // If "true" means that the sum is monotonic.
+    bool is_monotonic = 3;
+
+    // TODO: Decide if knowing that the metric values are aggregated from raw
+    // measurements is important.
+  }
 
-    // DOUBLE values are double-precision floating-point numbers.
+  // Represents the type of a metric that is calculated by aggregating as a
+  // Histogram of all reported measurements over a time interval.
+  //
+  // A Metric of this Type MUST store its values as HistogramDataPoint.
+  message Histogram {
+    // It describes the value type of the measurement used to build this
+    // aggregation.
     //
-    // A Metric of this Type MUST store its values as DoubleDataPoint.
-    DOUBLE = 3;
+    // Determines the value type of the exemplars.
+    MeasurementValueType measurement_value_type = 1;
+
+    // Temporality is the temporal quality values of a metric have.
+    Temporality temporality = 2;
 
-    // MONOTONIC_DOUBLE values are monotonically increasing double-precision
-    // floating-point numbers.
+    // TODO: Decide if knowing that the metric values are aggregated from raw
+    // measurements is important.
+
+    // TODO: Decide if knowing that the Sum is monotonic is important or not,
+    // may be derived from the buckets definition.
+  }
+
+  // Represents the type of a metric that is calculated by computing a summary
+  // of all reported measurements over a time interval.
+  //
+  // A Metric of this Type MUST store its values as SummaryDataPoint.
+  message Summary {
+    // It describes the value type of the measurement used to build this
+    // aggregation.
     //
-    // A Metric of this Type MUST store its values as DoubleDataPoint.
-    MONOTONIC_DOUBLE = 4;
-
-    // Histogram measurement.
-    // Corresponding values are stored in HistogramDataPoint.
-    HISTOGRAM = 5;
-
-    // Summary value. Some frameworks implemented Histograms as a summary of observations
-    // (usually things like request durations and response sizes). While it
-    // also provides a total count of observations and a sum of all observed
-    // values, it calculates configurable quantiles over a sliding time
-    // window.
-    // Corresponding values are stored in SummaryDataPoint.
-    SUMMARY = 6;
+    // Determines the value type of the exemplars.
+    MeasurementValueType measurement_value_type = 1;
+
+    // Temporality is the temporal quality values of a metric have.
+    Temporality temporality = 2;
+
+    // TODO: Decide if knowing that the metric values are aggregated from raw
+    // measurements is important.
+
+    // TODO: Decide if knowing that the Sum is monotonic is important or not,
+    // may be derived from the buckets definition.
   }
 
-  // type is the type of values this metric has.
-  Type type = 4;
+  // Type is the type of the metric and determines what are the type of the
```

Comment:
Can we make the three different uses of the word "type" here more explicit about what "type" they describe?

The last use is the one I'm confused on its meaning currently.

Reply:
Tried a new comment, please check.

Reply:
Looks great :+1:, thanks.